### PR TITLE
Adding examples to label epilepsy patients electrodes

### PR DESCRIPTION
**Pull request by tpatpa**
For iEEG data, we propose to describe the final conclusion drawn by an epileptologist on the electrodes involved in the seizures in the _electrodes.tsv. The final conclusion for each channel will be described in an additional column - seizure_zone, following HED format.

Please review @dorahermes

### CODE REVIEW

File: templates/sub-01/ses-01/ieeg/sub-epilepsy01_ses-01_electrodes.tsv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+name	x	y	z	Size    seizure_zone
```
Comment (dorahermes) [inline templates/sub-01/ses-01/ieeg/sub-epilepsy01_ses-01_electrodes.tsv:1]: Size should not be capitalized

**Comment by dorahermes** [comment]
lgtm! @sappelhoff this example for electrode labeling in epilepsy has been reviewed by several neurologists from different large epilepsy sites, when should we merge this?

**Comment by sappelhoff** [comment]
Do `electrodes.json` files even pass the validator? 🤔 I thought that the JSON file for electrodes.tsv is the coordsystem.json?

**Comment by dorahermes** [comment]
I think @dvanblooijs has done this and I believe it passed the validator. 

I remember it was a design decision to name _coordinatesystem.json different than _electrodes to ensure that _electrodes.json could be used for these purposes per the general spec that every .tsv file could have an accompanying .json.

**Comment by dvanblooijs** [comment]
If I read the manual correctly: coordystem.json is another file describing the location of fiducials, anatomical landmarks and coordinate system. We used the electrodes.json file to describe extra fields in our table that are not present in the BIDS version, like DKT atlases, resected area, and seizure onset zone.  It passes the validator if you locate the electrodes.json file in the same folder as the electrodes.tsv. When you don´t have an electrodes.json but you do have extra fields in your electrodes.tsv, you will get warnings that some fields are not described in electrodes.json.

**Comment by sappelhoff** [comment]
Thanks for trying this out @dvanblooijs  and for your explanations @dorahermes and @dvanblooijs.

I have looked into how this is implemented in the validator, and opened this issue: https://github.com/bids-standard/bids-validator/issues/1118

Basically, the support for this is there in both the spec and the validator. We just need to slightly adjust the validator to also allow electrodes.tsv+JSON at the "top" level and the "session" level ... mainly for consistency reasons. I don't think specifying electrods.tsv/json at the top level makes sense (however, specifying at the session level *does* make sense).

So that something like this would be possible (session level):

```
README
CHANGES
participants.tsv
sub-01/
    ses-01/
        sub-01_ses-01_electrodes.tsv
        sub-01_ses-01_electrodes.json
        sub-01_ses-01_coordsystem.json
        ieeg/
            sub-01_ses-01_task-rest_ieeg.edf
            sub-01_ses-01_task-rest_ieeg.json
            sub-01_ses-01_task-rest_channels.tsv
```

or this (top level):


```
README
CHANGES
participants.tsv
task-rest_electrodes.tsv
task-rest_electrodes.json
task-rest_coordsystem.json
sub-01/
    ses-01/
        ieeg/
            sub-01_ses-01_task-rest_ieeg.edf
            sub-01_ses-01_task-rest_ieeg.json
            sub-01_ses-01_task-rest_channels.tsv
```

File: templates/sub-01/ses-01/ieeg/sub-epilepsy01_ses-01_electrodes.tsv (hunk 1 above)
Comment (sappelhoff) [reply templates/sub-01/ses-01/ieeg/sub-epilepsy01_ses-01_electrodes.tsv:1]:
```suggestion
name	x	y	z	size    seizure_zone
```

**Comment by dvanblooijs** [comment]
I totally agree that an electrodes.json at session level or at top level makes sense! We had that previously but this gave an error when running the bids validator. I had the same problem with the events.json at top level. I asked this on neurostars (https://neurostars.org/t/quick-validation-failed-the-general-folder-structure-does-not-resemble-a-bids-dataset/2416/13), because the bids validator states that events.json could be at top level, but it gave an error when it was located at top level. Maybe this could be added to this issue?

**Comment by sappelhoff** [comment]
> We had that previously but this gave an error when running the bids validator. I had the same problem with the events.json at top level.

There is also some discussion on this here: https://github.com/bids-standard/bids-specification/issues/528 and in the PRs/issues linked from there.

I think the regular expressions currently cover more than 90% of valid use cases ... but there are still some use cases that are not appropriately covered. It'd need a really good look at it, or stepwise improvements. Feel free to suggest such improvements as issues to bids-validator like I just did in https://github.com/bids-standard/bids-validator/issues/1118 :-)

**Comment by dorahermes** [comment]
I will leave this issue open until it is officially resolved in the validator.

**Comment by sappelhoff** [comment]
> Thank you all for the inputs, I will leave this issue open until it is officially resolved in the validator.

I finally found a minute to adjust the validator with this PR: https://github.com/bids-standard/bids-validator/pull/1142

let's wait until it's merged, then we can merge this.

**Comment by Remi-Gau** [comment]
> > Thank you all for the inputs, I will leave this issue open until it is officially resolved in the validator.
> 
> I finally found a minute to adjust the validator with this PR: [bids-standard/bids-validator#1142](https://github.com/bids-standard/bids-validator/pull/1142)
> 
> let's wait until it's merged, then we can merge this.

as the PR on the validator got merge I will also merge this one.

**Comment by sappelhoff** [comment]
Thanks @Remi-Gau, I completely forgot :face_with_head_bandage:

**Pull request merged**